lib/Sema/SemaDecl.cpp

```diff
@@ -11262,6 +11270,100 @@ void Sema::AddInitializerToDecl(Decl *RealDecl, Expr *Init, bool DirectInit,
   CheckCompleteVariableDeclaration(VDecl);
 }
 
+/// Checked C: Validate that the NT_CHECKED type array initializers must be null terminated
```

Comment:
Please add some comments explaining the arguments more: Init is an init expression for a declaration with VDeclType.